core/src/main/java/com/kin/ecosystem/core/data/offer/OfferRepository.java

```diff
@@ -91,12 +99,55 @@ public void onFailure(ApiException e) {
 
 	private OfferList getList() {
 		OfferList masterList = new OfferList();
-		masterList.addAll(nativeOfferList);
-		masterList.addAll(cachedOfferList);
+
+		Offer tutorial = getTutorialOffer(cachedOfferList);
+		// Tutorial offer should be first
+		if (tutorial != null) {
+			masterList.add(tutorial);
+		}
+
+		Iterator<Offer> nativeSpendItr = nativeSpendOfferList.getOffers().iterator();
+		Iterator<Offer> nativeEarnItr = nativeEarnOfferList.getOffers().iterator();
+		Iterator<Offer> earnMPItr = cachedOfferList.getOffers().iterator();
+
+		while (nativeSpendItr.hasNext() || nativeEarnItr.hasNext() || earnMPItr.hasNext()) {
+			// 1st nativeSpendItr, 2ns nativeEarnItr
+			if(nativeSpendItr.hasNext()) {
```

Comment:
isnt it clearer to read 2 separate loops for each type then one loop

Reply:
No, the order is as follows: nativeSpendItr, nativeEarnItr, nativeSpendItr, nativeEarnItr.
and not all nativeSpendItr and then all nativeEarnItr.